src/app/clusters/bindings/BindingManager.h

```diff
@@ -123,22 +121,46 @@ class BindingManager
     static BindingManager & GetInstance() { return sBindingManager; }
 
 private:
-    static BindingManager sBindingManager;
-
-    static void HandleDeviceConnected(void * context, Messaging::ExchangeManager & exchangeMgr, SessionHandle & sessionHandle);
-    void HandleDeviceConnected(Messaging::ExchangeManager & exchangeMgr, SessionHandle & sessionHandle);
+    class ConnectionCallback
```

Comment:
@mrjerryjohns  This is the section that got added after your initial approval, please take a look and resolve if you have no further comments and are okay with the PR going in